server/lib/konnector_poller.coffee

```diff
@@ -18,187 +18,154 @@ month = 30 * day
-                    lastAutoImport = moment(konnector.lastAutoImport)
-
-                    # if we missed an importation cycle
-                    if (now.valueOf() - lastAutoImport.valueOf()) > importInterval
-                        log.debug "#{konnector.slug} missed an importation cycle"
-
-                        # calculate the supposed last Auto-import
-                        importTime = lastAutoImport.valueOf() + importInterval
-
-                        # calculate the time elapsed
-                        while importTime < now.valueOf()
-                            importTime += importInterval
-                        interval = importTime - now.valueOf()
-
-                        # We import now
-                        importer konnector
-
-                        nextUpdate = now.clone()
-                        nextUpdate = nextUpdate.add interval, 'ms'
-                        log.debug "#{konnector.slug} | Next update : " +
-                        "#{nextUpdate.format(format)}"
-
-                        @prepareNextCheck konnector, interval
+        if reset
+            nextUpdates = {}
+        if Object.keys(nextUpdates).length is 0
+            Konnector.all (err, konnectors) =>
+                async.eachSeries konnectors, (konnector, callback) =>
+                    # if both importInterval and lastAutoImport are valid
+                    if konnector.importInterval? \
+                    and konnector.importInterval isnt 'none' \
+                    and konnector.lastAutoImport? \
+                    and fs.existsSync path.resolve("server/konnectors/#{konnector.slug}.coffee")
+                        nextUpdate = @findNextUpdate(konnector)
+                        nextUpdates[konnector.slug] = [nextUpdate, konnector]
                         callback()
-
-                    # if we didn't missed an import interval
                     else
-                        #interval = (lastAutoImport + importInterval) - now
-                        interval = (lastAutoImport.valueOf() + importInterval)
-                        interval -= now.valueOf()
-                        log.debug "#{konnector.slug} didn't miss an importation cycle"
-
-                        nextUpdate = now.clone()
-                        nextUpdate = nextUpdate.add interval, 'ms'
-                        log.debug "#{konnector.slug} | Next update : " +
-                        "#{nextUpdate.format(format)}"
-
-                        @prepareNextCheck konnector, interval
                         callback()
-                else
-                    callback()
-
-    handleTimeout: (konnector) ->
-
-        # if date is present in fieldValues
+                , (err) =>
+                    # Initialize every day to avoid to have lots of long timeout.
+                    @prepareNextCheck()
+                    if timeout?
+                        clearTimeout timeout
+                    timeout = setTimeout @start.bind(@), day
```

Comment:
I think the poller timeout management could be extracted to a function, to avoid duplication and to make it more semantic/understandable. Also that will shorten this function.